### PR TITLE
Add SerialDisposable

### PR DESCRIPTION
**Pull request by nikolaykasyanov**
Hey, first of all, thanks for this awesome library.

I thought it'd be nice to have this type of disposable inspired by [RxSwift][rxswift] mostly, however I'm open to API and implementation suggestions/guidance.

[rxswift]: https://github.com/ReactiveX/RxSwift/blob/b4e4ea9b238a868b40b4951434f38a25c6c1b8ab/RxSwift/Disposables/SerialDisposable.swift

### CODE REVIEW

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/SerialDisposable.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.badoo.reaktive.utils.atomic.update
+
+/**
+ * Disposable whose underlying disposable can be replaced, causing this underlying resource's disposal.
+ *
+ * When [SerialDisposable] is disposed, it's underlying disposable is disposed and all disposables assigned
+ * after that are disposed immediately.
+ */
+class SerialDisposable : Disposable {
+    private val current = AtomicReference<Disposable?>(Disposable())
```
Comment (nikolaykasyanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/SerialDisposable.kt:13]: I noticed that `CompositeDisposable` is implemented with `synchronized` on JVM, unlike on native, please let me know if I should do the same here.

**Comment by arkivanov** [comment]
Hi, thanks for the PR! Isn't it the same as the [DisposableWrapper](https://github.com/badoo/Reaktive/blob/master/reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/DisposableWrapper.kt)?

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/SerialDisposable.kt (hunk 1 above)
Comment (arkivanov) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/SerialDisposable.kt:13]: Native `CompositeDisposable` is synchronized using atomics, same as `DisposableWrapper`.

**Comment by nikolaykasyanov** [comment]
Yeah looks like it, could find it due to unusual naming. Gonna close this one then, thanks!

**Comment by arkivanov** [comment]
> Yeah looks like it, could find it due to unusual naming. Gonna close this one then, thanks!

Thanks for the feedback, I will consider renaming.

**Comment by arkivanov** [comment]
Filed #598

**Pull request closed without merging**